src/storage/txn/mod.rs

```diff
@@ -36,7 +45,65 @@ quick_error! {
             cause(err)
             description(err.description())
         }
+        Other(err: Box<error::Error + Sync + Send>) {
+            from()
+            cause(err.as_ref())
+            description(err.description())
+            display("{:?}", err)
+        }
+        Io(err: IoError) {
+            from()
+            cause(err)
+            description(err.description())
+        }
     }
 }
 
 pub type Result<T> = ::std::result::Result<T, Error>;
+
+
+// send_msg wraps Sender and retries some times if queue is full.
+pub fn send_msg<M: Send>(ch: &mio::Sender<M>, mut msg: M) -> Result<()> {
+    for _ in 0..MAX_SEND_RETRY_CNT {
+        let r = ch.send(msg);
+        if r.is_ok() {
+            return Ok(());
+        }
+
+        match r.unwrap_err() {
+            NotifyError::Full(m) => {
+                warn!("notify queue is full, sleep and retry");
+                thread::sleep(Duration::from_millis(100));
```

Comment:
@BusyJay This function will be called by raftstore thread. Is it okay to make raftstore thread sleep if scheduler's notify channel is full?